Sources/ClientRuntime/Interceptor/AnyInterceptor.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Type-erased, concrete interceptor.
+/// 
+/// In order to have multiple interceptors hooked into a single operation, we
+/// need a concrete type, not a protocol. This stores references to the closures
+/// of interceptor implementations and delegates to them for each interceptor hook.
+internal struct AnyInterceptor<RequestType, ResponseType, AttributesType: HasAttributes> {
```

Comment:
Should `AnyInterceptor` conform to `Interceptor`?

Reply:
also, nit: Swift types are `internal` by default, and the `internal` keyword is usually only applied where the compiler infers otherwise (i.e. in a `public` or `private` extension.)

Reply:
Good question - `AnyInterceptor` can't because it must operation on `DefaultInterceptorContext` with concrete `ResponseType` etc, which can't be guaranteed by the `Interceptor` protocol.